clang/lib/Sema/SemaTemplateInstantiate.cpp

```diff
@@ -1841,9 +1872,71 @@ Decl *TemplateInstantiator::TransformDecl(SourceLocation Loc, Decl *D) {
     // template parameter.
   }
 
+  if (auto *PVD = dyn_cast<ParmVarDecl>(D);
+      PVD && PVD->getDeclContext() == FunctionDCForParameterDeclInstantiation &&
+      !ParameterInstantiationScope->findInstantiationUnsafe(PVD)) {
+    // Fall through to the default lookup even if we have failed to instantiate
+    // anything. We're likely to crash thereafter.
+    instantiateParameterToScope(PVD, *ParameterInstantiationScope);
+  }
+
   return SemaRef.FindInstantiatedDecl(Loc, cast<NamedDecl>(D), TemplateArgs);
 }
 
+bool TemplateInstantiator::instantiateParameterToScope(
```

Comment:
Can we reuse the logic of `SetupConstraintScope` / `addInstantiatedParametersToScope` ? It seems very similar